### PR TITLE
Bonnie v3.1 composite excel patient sharing

### PR DESCRIPTION
**Pull request by daco101**
This PR adds support for sharing patients across all components of a composite measure and the composite measure itself, and fixes excel exports for shared patients and composite measures. It adds a 'patient is shared' warning to the patient builder, and it disables patient creation at the composite measure.
Tests are included for both front end and back end excel exports.

Note: Excel exports sometimes have the observ mismatch issue (like expected is null, actual is 0, or expected is [nil], actual []) so the coloring in the excel can be off.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR:https://jira.mitre.org/browse/BONNIE-1755, https://jira.mitre.org/browse/BONNIE-1754
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] NA If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 

Branch | Back End Coverage | Front End Coverage
-- | -- | --
bonnie-v3.1 | 2205 / 2458 LOC (89.71%) covered | Statements   : 59.61% ( 7318/12277 )<br>Branches     : 43.55% ( 1991/4572 )<br>Functions    : 50.1% ( 1216/2427 )<br>Lines        : 57.51% ( 6574/11431 )
bonnie-v3.1_composite_excel_patient_sharing | 2219 / 2472 LOC (89.77%) covered | Statements   : 59.65% ( 7358/12336 )<br>Branches     : 43.72% ( 2005/4586 )<br>Functions    : 50.27% ( 1225/2437 )<br>Lines        : 57.53% ( 6605/11481 )

- [x] NA Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] NA JIRA test links:
- [x] NA Justification for using JIRA tests:
- [x] NA JIRA tests have been added to sprint

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree N/A

### CODE REVIEW

**Comment by lukoktonos** [comment]
There seem to be some new errors and warnings that don't make the tests fail, but still would be best to not have if possible:

`TypeError: undefined is not an object (evaluating 'specific_value_set.concepts')`

Multiple `WARNING: missing value set`s

Have you looked into what is causing these?

**Review by lukoktonos — APPROVED** [review]
Looks good--great work

File: lib/patient_export.rb [hunk 1]
```diff
@@ -138,6 +138,11 @@ def self.export_excel_cql_file(calc_results, patient_details, population_details
             population_details[pop_key]["statement_relevance"].each do |lib_key, statements|
               statements.each do |statement, relevance|
                 if (relevance != "NA")
+                  # Composite measures will have some statement results from the components, but
+                  # the measure won't have statement details (excel headers) for them. 
+                  # Fortunately we want to ignore those results, so we can just skip results that
+                  # dont have corresponding details from the measure.
+                  next if statement_details.dig(lib_key,statement).nil?
```
Comment (jbradl11) [inline lib/patient_export.rb:145]: Neat!

Comment (lukoktonos) [reply lib/patient_export.rb:145]: I dig it.

Comment (daco101) [reply lib/patient_export.rb:145]: :pick:

**Review by jbradl11 — APPROVED** [review]

**Pull request merged**